File: api/src/monster/types/movement.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, PartialEq, sqlx::Type, Serialize, Deserialize)]
#[sqlx(type_name = "movement_type", rename_all = "snake_case")]
#[serde(rename_all = "snake_case")]
pub enum Movement {
    Normal,
    Burrow,
    Climb,
    Fly,
    FlyHover,
    Swim,
}

File: view/src/lib/types/Movement.ts
export enum Movement {
  Normal = "normal",
  Burrow = "burrow",
  Climb = "climb",
  Fly = "fly",
  FlyHover = "fly_hover",
  Swim = "swim",
}

export const DisplayMovement: Readonly<Record<Movement, string>> = {
  [Movement.Normal]: "Normal",
  [Movement.Burrow]: "Burrow",
  [Movement.Climb]: "Climb",
  [Movement.Fly]: "Fly",
  [Movement.FlyHover]: "Fly (hover)",
  [Movement.Swim]: "Swim",
} as const;

export const Movements: Movement[] = Object.values(Movement);
